Npc's

James120393 opened the pull request.

NecroReindeer committed:
Created project starting point

Created project starting point that displays empty window.
Documented using doxygen compatible comments
http://www.stack.nl/~dimitri/doxygen/manual/docblocks.html

NecroReindeer committed:
Delete ReadMe.txt

NecroReindeer committed:
Added template for texture and game objects

James120393 committed:
Added base classes for NPC's

NecroReindeer committed:
Made window display a grid of cells.

Set up level and cell classes and display a grid of cells

NecroReindeer committed:
Fixed Level destructor and documented code

Documented Level and LevelCell.
Fixed potential problem with destructor of Level.

NecroReindeer committed:
Merge pull request #1 from NecroReindeer/level-generation

Merge basic level generation grid classes into base

NecroReindeer committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into level-generation

NecroReindeer committed:
Set up CellEdge classes and render walls

CellWalls are now rendered

NecroReindeer committed:
Fixed bug with wall directions.

South and north walls are not in the correct place.

NecroReindeer committed:
Implemented growing tree algorithm to generate maze

NecroReindeer committed:
Added doors and fixed LevelCell destructor

NecroReindeer committed:
Added rooms

Maze now consists of randomly shaped rooms

NecroReindeer committed:
Create CHANGELOG.md

NecroReindeer committed:
Create CHANGELOG.md

NecroReindeer committed:
Delete CHANGELOG.md

NecroReindeer committed:
Tidied code in level class

NecroReindeer committed:
Update CHANGELOG.md

NecroReindeer committed:
Coding Task 2 Proposal

NecroReindeer committed:
(Add trello board screenshot for COMP110 Proposal)

NecroReindeer committed:
Merge branch 'level-generation' of https://github.com/NecroReindeer/comp150-desktop-game into level-generation

NecroReindeer committed:
Added COMP110 Trello board screenshot

NecroReindeer committed:
Added exit and made it be placed on right/top edge

NecroReindeer committed:
Made LevelCell and Exit shared pointers in Level

NecroReindeer committed:
Update README.md

NecroReindeer committed:
Made CellEdges use shared pointers

NecroReindeer committed:
Tidied CellEdge classes and GameObject class

CellEdge, CellWall, and CellPassage now documented.
GameObject no longer duplicates code from Texture.

NecroReindeer committed:
Made Level use shared pointers for rooms

NecroReindeer committed:
Made level bigger and used smaller sprites

NecroReindeer committed:
Merge pull request #2 from NecroReindeer/level-generation

Level generation basic maze

James120393 committed:
Haven't changed much if anything.

James120393 committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into NPC's

NecroReindeer committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into level-generation

James120393 committed:
Added Doctors & Guards Pos

NecroReindeer committed:
Update CHANGELOG.md

James120393 committed:
Added Player Class and Documented

Inherits From Game

James120393 committed:
Fixed Player Class Bugs

NecroReindeer committed:
Merge pull request #3 from NecroReindeer/NPC's

Npc s

James120393 committed:
Added Player start position.

James120393 committed:
Put the Start code in the player class

NecroReindeer committed:
Merge pull request #4 from NecroReindeer/NPC's

Player start position

James120393 committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into NPC's

NecroReindeer committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into controls

NecroReindeer committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into level-generation

NecroReindeer committed:
Added basic player movement

NecroReindeer committed:
Merge pull request #5 from NecroReindeer/controls

Controls

NecroReindeer committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into level-generation

NecroReindeer committed:
Merge branch 'level-generation' of https://github.com/NecroReindeer/comp150-desktop-game into level-generation

NecroReindeer committed:
Made multiple doctors and guards in random pos

NecroReindeer committed:
Made docs and guards inherit from Character class

NecroReindeer committed:
Changed npc creation to use template function

Doctors and guards are now made using a template function
createCharacter, rather than in two separate methods

NecroReindeer committed:
Tidied up the new code

NecroReindeer committed:
Update CHANGELOG.md

NecroReindeer committed:
Update CHANGELOG.md

NecroReindeer committed:
Made all classes use new VectorXY for position

NecroReindeer committed:
Made sprite size in texture class use VectorXY

NecroReindeer committed:
Made player and player movement use VectorXY

NecroReindeer committed:
Doctors and Guards now use VectorXY for start pos

NecroReindeer committed:
Exit now uses VectorXY for initial position

NecroReindeer committed:
Made position be determined outside CreateCharacter

NecroReindeer committed:
Made Player inherit from Character and tidied code

NecroReindeer committed:
Update CHANGELOG.md

NecroReindeer committed:
Merge pull request #6 from NecroReindeer/refactoring

Refactoring - VectorXY and Character

James120393 committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into NPC's

NecroReindeer committed:
Updated changelog with other members' commits

James120393 committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into NPC's

NecroReindeer committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into level-generation

# Conflicts:
#	CHANGELOG.md

NecroReindeer committed:
Made NPCs not start in same position as player

NPCs now don't start in same position as player.
Level has a vector of characters instead of a vector of npcs.
As a consequence, player position is a static member of level, and now
level has a getter to give the game access to the player.

NecroReindeer committed:
Made level be regenerated when spacebar pressed

NecroReindeer committed:
Level restarts when player centre is at exit centre

NecroReindeer committed:
Added documentation to Level, CellEdge, and VectorXY

NecroReindeer committed:
Added documentation to VectorXY

NecroReindeer committed:
Merge pull request #7 from NecroReindeer/level-generation

NPCs in unique positions and new levels can be generated

James120393 committed:
Attempting to add movements

NecroReindeer committed:
Added methods to get cell edge and check if wall

Added a method to LevelCell to get a pointer to an edge in a given
direction.
Added a method to CellEdge and its subclasses to return as a bool
whether the edge is a wall or not.

NecroReindeer committed:
Added documentation to isWall() and getEdge()

NecroReindeer committed:
Added method to convert window to grid coords

NecroReindeer committed:
Merge pull request #8 from NecroReindeer/level-generation

Getting cell edges and checking walls

NecroReindeer committed:
Update CHANGELOG.md

NecroReindeer committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into level-generation

NecroReindeer committed:
Made Doors separate class and used template func

Made Doors a separate class, and used template function to initialise
cell edges, instead of having a separate function for each

NecroReindeer committed:
Added random chance for door to be locked

NecroReindeer committed:
Added experimental corridor code to level gen

James120393 committed:
Added NPC classes.

James120393 commented on doomed_patient/doomed_patient/Directions.cpp:
```diff
@@ -13,6 +13,8 @@ Directions::~Directions()
 
 VectorXY Directions::getDirectionVector(Direction direction)
 {
+	int X = rand() % 4;
```
Ignore this bit.

James120393 commented on doomed_patient/doomed_patient/Directions.cpp:
```diff
@@ -23,6 +25,8 @@ VectorXY Directions::getDirectionVector(Direction direction)
 	case Directions::Direction::WEST:
 		return VectorXY(-1, 0);
+	case Directions::Direction::RANDOM:
+		return VectorXY(X, Y);
```
And this.

James120393 commented on doomed_patient/doomed_patient/LevelCell.h:
```diff
@@ -67,15 +67,35 @@ class LevelCell : public GameObject, public std::enable_shared_from_this<LevelCe
 	*/
 	static const int NUMBER_OF_SIDES = 4;
 
+	struct Passable
```
Added this to check cells

James120393 commented on doomed_patient/doomed_patient/NPC.cpp:
```diff
@@ -0,0 +1,101 @@
+
+	//North -> South
+
+	if (movementDirection == Directions::Direction::NORTH && !blocked.NORTH)
```
First itteration of movement criteria.

James120393 commented on doomed_patient/doomed_patient/PatientGame.cpp:
```diff
@@ -108,7 +116,24 @@ void PatientGame::handleEvents()
 {
+	for each (std::shared_ptr<Character> npc in level.characters)
+	{
+		// TODO: Bug where centre appears to be checking wrong cell for north and west edges
```
Here it is

James120393 commented on doomed_patient/doomed_patient/PatientGame.cpp:
```diff
@@ -108,7 +116,24 @@ void PatientGame::handleEvents()
+			level.getCell(npc->getCentre().convertWindowToGrid())->getEdge(Directions::Direction::EAST)->isWall(),
+			level.getCell(npc->getCentre().convertWindowToGrid())->getEdge(Directions::Direction::SOUTH)->isWall(),
+			level.getCell((npc->getCentre()).convertWindowToGrid())->getEdge(Directions::Direction::WEST)->isWall()
+		};
```
As it scans the cell edge when looking for a clear passage when it scans the North and West edge the npc skips the entire cell when approaching from the South or East. I think its due to the way that the edges are currently set-up in my build. And I think thats explaining it properly but don't quote me on that.

NecroReindeer committed:
Experimenting with rooms

NecroReindeer commented on doomed_patient/doomed_patient/NPC.cpp:
```diff
@@ -0,0 +1,101 @@
+	}
+	else if (movementDirection == Directions::Direction::NORTH && blocked.NORTH && !blocked.SOUTH)
+	{
+		movementDirection = Directions::Direction::SOUTH;
```
Since a lot of these `if` statements result in a reversed direction, you might want to check out using `Directions::getOpposite(Direction direction)`

NecroReindeer committed:
Added methods to check if npc is past centre of cell

NecroReindeer committed:
Commented code

NecroReindeer committed:
Made seed be random again

NecroReindeer commented:
Made pull request with fix to your NPCs branch ^_^

NecroReindeer committed:
Fixed problem with weak/shared pointers to cell/room

NecroReindeer committed:
Moved door assignment to separate function

Moved door assignment to separate function and added sprite to visualise
corridors to aid testing

NecroReindeer committed:
Made it check min and max room size

NecroReindeer committed:
Moved maze generation to Maze class

NecroReindeer committed:
Removed clearMaze() method

James120393 committed:
Merge pull request #9 from NecroReindeer/Character-Cell-fix

Character cell fix

NecroReindeer committed:
Added class for managing maze generation

NecroReindeer committed:
Made MazeGenerationManager use getters

NecroReindeer committed:
Merge pull request #10 from NecroReindeer/refactoring

Refactoring

NecroReindeer committed:
Added method to make corridors straighter

NecroReindeer committed:
Made it always use biased direction for corridors

NecroReindeer committed:
Changed min/max room size

NecroReindeer committed:
Made it fullscreen again

NecroReindeer committed:
Merge pull request #11 from NecroReindeer/level-generation

Level generation

James120393 committed:
Removed unwanted code.

NecroReindeer committed:
Merge remote-tracking branch 'refs/remotes/origin/NPC's'

# Conflicts:
#	doomed_patient/doomed_patient/CellEdge.h
#	doomed_patient/doomed_patient/Level.cpp
#	doomed_patient/doomed_patient/Level.h
#	doomed_patient/doomed_patient/LevelCell.h
#	doomed_patient/doomed_patient/PatientGame.cpp
#	doomed_patient/doomed_patient/doomed_patient.vcxproj
#	doomed_patient/doomed_patient/doomed_patient.vcxproj.filters

NecroReindeer committed:
Fixed merge conflicts

James120393 committed:
Tried adding code to prevent NPC's from moving through doors (Not Working)

James120393 committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into NPC's

# Conflicts:
#	doomed_patient/doomed_patient/CellPassage.cpp
#	doomed_patient/doomed_patient/Level.h
#	doomed_patient/doomed_patient/PatientGame.cpp

These conflicts have been fixed.

James120393 committed:
Updated from master

Still not working, probably a simple mistake that I can not see.

James120393 committed:
NPC's not moving through door. The result that was wanted

James120393 committed:
Added Collision to the player, but introduced an issue with the keyboard detection update.

James120393 committed:
Re factored code in character to remove duplications.

Valryn560 committed:
Player moves a little less insanely, added a private movementcount as a controller for this then referred to it in Player.cpp

Valryn560 committed:
HALP!

James120393 committed:
Removed duplicate game

James120393 committed:
Further Removal of junk files.

James120393 committed:
Delete doomed_patient.sln

James120393 committed:
Hopefully Fixed all file issues.

James120393 committed:
Merge branch 'NPC's' of https://github.com/NecroReindeer/comp150-desktop-game into NPC's

# Conflicts:
#	doomed_patient/doomed_patient/doomed_patient.sln

#####SORTED######

Valryn560 committed:
Added code for review.

In Player.cpp

and NPC.cpp

Valryn560 committed:
Player Reset position fixed

James120393 committed:
Added Sprite Directions when changing direction.
